tools/python_tools/cuvslam_tools/tests/test_coda_conversion.py

```
@@ -0,0 +1,413 @@
+    def test_zero_disparity_entry_is_rejected(self):
+        _write_archive(self.raw_dir, poses=_default_poses(), intrinsics=_intrinsics_yaml(disparity_q14=0.0))
+
+        with self.assertRaisesRegex(convert_coda.ConversionError, "cannot derive baseline"):
+            _convert(self.raw_dir, self.output_dir)
+
+    def test_unterminated_data_list_is_reported(self):
+        # A truncated calibration file: the data list never reaches its "]".
+        truncated = _intrinsics_yaml().split("disparity_matrix")[0].rstrip("\n")[:-1]
+        _write_archive(self.raw_dir, poses=_default_poses(), intrinsics=truncated)
+
+        with self.assertRaisesRegex(
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)